src/bidiMapper/domains/events/SubscriptionManager.ts

```diff
@@ -158,33 +195,57 @@ export class SubscriptionManager {
     eventMap.set(event, this.#subscriptionPriority++);
   }
 
+  /**
+   * Unsubscribes atomically from all events in the given contexts and channel.
+   */
+  unsubscribeAll(
+    events: Session.SubscribeParametersEvent[],
+    contextIds: (CommonDataTypes.BrowsingContext | null)[],
+    channel: string | null
+  ) {
+    // Assert all contexts are known.
+    for (const contextId of contextIds) {
+      if (contextId !== null) {
+        this.#browsingContextStorage.getKnownContext(contextId);
+      }
+    }
```

Comment:
this check would logically fit better into `#checkUnsubscribe`

Reply:
It's more logical and it fits better, however the side effect is a small performance penalty. As of now:

- events: E1, E2
- contexts: C1, C2

Contexts are checked only twice: C1 and C2.

If we do what you propose, then it becomes:

[E1, C1], [E1, C2], [E2, C1], [E2, C2] --> call `checkUnsubscribe` for each of those --> Contexts are checked 4 times - 2 for C1 and 2 for C2.

If you don't care about the redundancy, then I'll go ahead.

Reply:
Actually, this change makes the unit test fail! https://github.com/GoogleChromeLabs/chromium-bidi/actions/runs/4325825970/jobs/7552477329

Reply:
Reverted this bit back to the original form.